domain/src/main/java/edu/nextstep/camp/domain/calculator/BinaryCalculator.kt

```diff
@@ -0,0 +1,31 @@
+package edu.nextstep.camp.domain.calculator
+
+import java.lang.IllegalArgumentException
+
+typealias CalculationStrategy = (Int, Int) -> Int
+
+enum class Formula(val operator: String, val formula: CalculationStrategy) {
+    PLUS("+", { x, y -> x + y }),
+    MINUS("-", { x, y -> x - y }),
+    MULTIPLY("×", { x, y -> x * y }),
+    DIVIDE("÷", { x, y -> x / y });
+
+    companion object {
+        fun matchFormula(token: String): CalculationStrategy = when(token) {
+            PLUS.operator -> PLUS.formula
+            MINUS.operator -> MINUS.formula
+            MULTIPLY.operator -> MULTIPLY.formula
+            DIVIDE.operator -> DIVIDE.formula
+            else -> throw IllegalArgumentException("사칙연산 기호가 아닙니다")
+        }
```

Comment:
```suggestion
        fun findBy(token: String): Formula {
            return values().find { it.operator == token }
                ?: throw IllegalArgumentException("$token 은 사칙연산 기호가 아닙니다")
        }
```
이렇게 토큰으로 Enum을 찾을 수 있습니다.

Fomula에는 멤버 함수로
```kotlin
fun calculate(left: Int, right:Int) : Int{
    return formula(left, right)
}
```
이렇게 갖는다면,

``` kotlin
val formula = Formula.findBy("+")
val result = formula.caculate(1, 2) // == 3
```

이런 식으로 사용할 수 있습니다 🙂